jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ChangeConnectionSettingsMenu.kt

```diff
@@ -1,232 +0,0 @@
-    }
-}
-
-internal class ChangePartitionActionGroup(private val accountSettingsManager: AwsConnectionManager) :
-    ComputableActionGroup(message("settings.partitions"), true), DumbAware {
-    override fun createChildrenProvider(actionManager: ActionManager?): CachedValueProvider<Array<AnAction>> = CachedValueProvider {
-        val selectedPartitionId = accountSettingsManager.selectedPartition?.id
-        val actions = AwsRegionProvider.getInstance().partitions().values.filter { it.id != selectedPartitionId }.map { partition ->
-            ChangeRegionActionGroup(partition, accountSettingsManager, name = partition.description)
-        } as List<AnAction>
-
-        CachedValueProvider.Result.create(actions.toTypedArray(), accountSettingsManager)
-    }
-}
-
-internal class ChangeRegionActionGroup(
-    private val partition: AwsPartition?,
-    private val accountSettingsManager: AwsConnectionManager,
-    private val partitionSelector: ChangePartitionActionGroup? = null,
-    name: String = message("settings.regions.region_sub_menu")
-) : ComputableActionGroup(name, true), DumbAware {
-    private val regionProvider = AwsRegionProvider.getInstance()
-
-    override fun createChildrenProvider(actionManager: ActionManager?): CachedValueProvider<Array<AnAction>> = CachedValueProvider {
-        val (regionMap, partitionGroup) = partition?.let {
-            // if a partition has been selected, only show regions in that partition
-            // and the partition selector
-            regionProvider.regions(partition.id) to partitionSelector
-            // otherwise show everything with no partition selector
-        } ?: regionProvider.allRegions() to null
-
-        val actions = mutableListOf<AnAction>()
-
-        regionMap.values.groupBy { it.category }.forEach { (category, subRegions) ->
-            actions.add(Separator.create(category))
-            subRegions.forEach {
-                actions.add(ChangeRegionAction(it))
-            }
-        }
-
-        if (partitionGroup != null && regionProvider.partitions().size > 1) {
-            actions.add(Separator.create())
-            actions.add(partitionGroup)
-        }
-
-        CachedValueProvider.Result.create(actions.toTypedArray(), accountSettingsManager)
-    }
-}
-
-internal class ChangeRegionAction(private val region: AwsRegion) : ToggleAction(region.displayName), DumbAware {
-    override fun isSelected(e: AnActionEvent): Boolean = getAccountSetting(e).selectedRegion == region
-
-    override fun setSelected(e: AnActionEvent, state: Boolean) {
-        if (state) {
-            val oldRegion = getAccountSetting(e).selectedRegion
-
-            getAccountSetting(e).changeRegion(region)
-
-            if (oldRegion?.partitionId != region.partitionId) {
-                AwsTelemetry.setPartition(
-                    partitionId = region.partitionId
-                )
-            }
-
-            AwsTelemetry.setRegion(e.project)
```

Comment:
need these back

Reply:
Yikes...good catch

Reply:
Moved logic into AwsConnectionManager which matches where we do the validate telemetry